### PR TITLE
fixes issues #11988 with corrupted json

### PR DESCRIPTION
**Pull request by kr-g**
i added a simple counter.
since i don't wanted to spent too much time investigating on how stack unwinding works (lots of goto's)

### CODE REVIEW

File: extmod/modjson.c [hunk 1]
```diff
@@ -157,6 +157,7 @@ STATIC mp_obj_t mod_json_load(mp_obj_t stream_obj) {
     const mp_obj_type_t *stack_top_type = NULL;
     mp_obj_t stack_key = MP_OBJ_NULL;
     S_NEXT(s);
+    unsigned int nested_cnt = 0;
     for (;;) {
     cont:
         if (S_END(s)) {
```
Comment (kr-g) [inline extmod/modjson.c:163]: i seems that this is not reached. adding the nested_cnt check directly before label success was never reached (at least with my buggy json)

**Comment by kr-g** [comment]
@jimmo please advise with regards to code formatting etc

**Comment by kr-g** [comment]
looks like that because i have installed pre-commit too late i do have now some "wrong" commit messages in my history preventing the build from succeeding...

**Comment by kr-g** [comment]
rebased (reword) the commit messages and signed all off
removed last merge (from master) from the branch

**Comment by dlech** [comment]
Even better would be to squash them all into one commit since this is really one change/fix.

**Comment by kr-g** [comment]
> Even better would be to squash them all into one commit since this is really one change/fix.

well it is quite time consuming to contribute here.
and not that easy to find out what needs to be done to fulfill all requirements for all steps.
the efforts for cotribution and the proces is high and documentation missing or scattered, or just available in the sense of a best guess.

until there is no further check (in the whole process) what prevents to have more that 1 commit in 1 pull request - i m fine with how it is right now.

i dont want to spent more time on that issue. looking back now on the efforts for that kind of bureaucracy i would not have started to contribute (if i had known before...)

**Comment by kr-g** [comment]
but probably there is a good reason why it is like that

**Comment by kr-g** [comment]
due to lack of time i m closing this.
feel free to merge or start from scratch with another solution,
or leave it out completely by decision.

**Pull request closed without merging**